WorkBranch

JiaoJiao123 opened the pull request:
1: use , as and, use ; as or and will read the expression from left to
right
2: update about mouse over highlight feature
1) remove the previously highlighted biomarkers and   only
highlight the mouse over biomarker
2) removed the alert box

JiaoJiao123 committed:
highlight biomarker feature added and fix bug upon confirm alteration

1: upon mouse over, an alert box will pop up to show how many matches
are found. and the corresponding matches will be highlighted in the text
2: before after curator edit an alteration and refresh the page, the
original alterations will still be displayed. This bug get fixed
3: once curator did any curation, the curation status will
automatically change to curating.
4: in the curation history, the verbs are changed to past sense and the
time format are changed to the user friendly reading way.

JiaoJiao123 committed:
search engine update

1: use , as and, use ; as or and will read the expression from left to
right
2: update about mouse over highlight feature
    1) remove the previously highlighted biomarkers and   only
highlight the mouse over biomarker
    2) removed the alert box

zhx828 commented on public/modules/core/controllers/home.client.controller.js:
```diff
@@ -144,29 +145,30 @@ angular.module('core').controller('HomeController', ['$scope', '$location', '$ro
                             _.each(trials[i].tumorTypes, function (tumorItem) {
                                 $scope.tumorTypes.push(tumorItem.tumorTypeId);
                             });
+                            if (alterationsFetched[i] !== undefined) {
```
Do you wanna check whether alterationsFetched is Array first?

zhx828 commented on public/modules/core/controllers/home.client.controller.js:
```diff
@@ -144,29 +145,30 @@ angular.module('core').controller('HomeController', ['$scope', '$location', '$ro
                                 $scope.tumorTypes.push(tumorItem.tumorTypeId);
                             });
+                            if (alterationsFetched[i] !== undefined) {
+                                if (alterationsFetched[i].alterationsFetched.length > 0) {
```
The same here, alterationsFetched[i]. alterationsFetched wont always be Array, right?

JiaoJiao123 committed:
system test and bug fix

1: double checked biomarker prediction procedure, fix the bugs and
improved prediction precision
2: once any alteration operation happened, now will refetch from db
instead of change $scope.trialAlterations to change the page content.
By doing this, get rid of the problem of returning alterionID when
adding new alteration.
3: fix the bug about the highlight biomarkers.

JiaoJiao123 committed:
highlight feature function call change

call highlight functions with an elementID array instead of call by
elementID one by one

JiaoJiao123 committed:
add condition check for string input in the highlight function

if elementId input is string, then transfer it to array
